arwen/elrondapi/elrondei.go

```diff
@@ -343,32 +349,32 @@ func blockHash(context unsafe.Pointer, nonce int64, resultOffset int32) int32 {
 	//TODO: change blockchain hook to treat actual nonce - not the offset.
 	hash := blockchain.BlockHash(nonce)
 	err := runtime.MemStore(resultOffset, hash)
-	if withFault(err, context) {
+	if arwen.WithFault(err, context) {
 		return 1
 	}
 
 	return 0
 }
 
 //export transferValue
-func transferValue(context unsafe.Pointer, gasLimit int64, destOffset int32, valueOffset int32, dataOffset int32, length int32) int32 {
+func transferValue(context unsafe.Pointer, destOffset int32, valueOffset int32, dataOffset int32, length int32) int32 {
 	runtime := arwen.GetRuntimeContext(context)
 	metering := arwen.GetMeteringContext(context)
 	output := arwen.GetOutputContext(context)
 
 	send := runtime.GetSCAddress()
 	dest, err := runtime.MemLoad(destOffset, arwen.AddressLen)
-	if withFault(err, context) {
+	if arwen.WithFault(err, context) {
```

Comment:
there is no need to return anything from such a function if we get out from it with a breakpoint. so the decision has to be taken, whether to not return anything, and get out instantenausly from the wasmer context with error. Or leave the SC developer to do whatever he wants if a transfer call failed.

Reply:
Changed `WithError` to be more flexible. It may (or may not) set a breakpoint depending on the internal state of the Runtime context.